include/aspect/material_model/depth_dependent.h

```diff
@@ -0,0 +1,183 @@
+         * These are derived from the base model.
+         */
+        virtual bool
+        viscosity_depends_on (const NonlinearDependence::Dependence dependence) const;
+
+        /**
+         * Method to determine density dependencies of material model
+         * These are derived from the base model.
+         */
+        virtual bool
+        density_depends_on (const NonlinearDependence::Dependence dependence) const;
+
+        /**
+         * Method to determine compressibility dependencies of material model
+         * These are derived from the base model.
+         */
+        virtual bool
+        compressibility_depends_on (const NonlinearDependence::Dependence dependence) const;
+
+        /**
+         * Method to determine specific heat capacity dependencies of material model
+         * These are derived from the base model.
+         */
+        virtual bool
+        specific_heat_depends_on (const NonlinearDependence::Dependence dependence) const;
+
+        /**
+         * Method to determine thermal conductivity dependencies of material model
+         * These are derived from the base model.
+         */
+        virtual bool
+        thermal_conductivity_depends_on (const NonlinearDependence::Dependence dependence) const;
+
+        /**
+         * Method that indicates whether material is compressible. Depth dependent model is compressible
+         * if and only if base model is compressible.
+         */
+        virtual bool is_compressible () const;
+
+        /**
+         * Method to calculate reference viscosity for the depth-dependent model. The reference
+         * viscosity is determined by evaluating the depth-dependent part of the viscosity at
+         * the mean depth of the model.
+         */
+        virtual double reference_viscosity () const;
+
+        /**
+         * Method to calculate reference density for the depth-dependent model. Because the depth-
+         * dependent model deos not modify density, the reference density is equivalent to the
+         * base model's reference density.
+         */
+        virtual double reference_density () const;
+
+      private:
+        enum
+        {
+          Function,
+          File,
+          List,
+          None
+        } viscosity_method;
+
+        /**
+         * Function to read depth-dependent lookup table and set up interpolating function
+         * for File depth dependence method
+         */
+        void
+        read_viscosity_file(const std::string &filename);
+
+        /**
+         * Data structures to store depth and viscosity lookup tables as well as interpolating
+         * function to calculate viscosity for File Depth dependence method
+         */
+        std_cxx11::array< std::vector<double>, 1 > depth_table;
+        std_cxx1x::shared_ptr<Table<1,double>> viscosity_table;
+        std_cxx1x::shared_ptr<Functions::InterpolatedTensorProductGridData<1>> f;
```

Comment:
the correct spelling for the namespace is now `std_cxx11`. In the second and third line, you will need a space between the closing `> >` to make older compilers happy.

Would you consider using a less generic variable name than `f`? It doesn't really give away what its purpose is.